packages/block/src/blockProcessor.ts

```diff
@@ -66,7 +67,16 @@ export const blockFactory = (provider: ethers.providers.Provider) => async (bloc
             hash: block.hash,
             number: block.number,
             parentHash: block.parentHash,
-            transactions: transactions,
+            transactions: transactions.map(tx => ({
+                nonce: tx.nonce,
+                blockNumber: tx.blockNumber,
+                to: tx.to,
+                from: tx.from,
+                chainId: tx.chainId,
+                data: tx.data,
+                value: tx.value.toString(),
```

Comment:
Shouldnt toString here

Reply:
as discussed above, I left the whole Block hierarchy as plain objects, so they are guaranteed not to need serialisation; so I stringify the only two BigNumbers here.